### ISSUE TITLE
Return type incorrectly inferred as `Unknown` when no return statement is present

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3859]
```python
def func(self, verts, closed):
    if not closed:
        return
    if isinstance(verts, list) and len(verts):
        verts_pad = np.concatenate()
        codes = np.empty()
    for xy in verts:
        if len(xy):
            if isinstance(xy, int):
                pass
```

**Comment by erictraut** [comment]
Commit: https://github.com/microsoft/pyright/commit/41c47a08ea496029b9338df508ea36e7bdc0ed01

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.268, which I just published. It will also be included in a future version of pylance.
